shotgun_api3/shotgun.py

```diff
@@ -1752,8 +1794,24 @@ def _response_errors(self, sg_response):
         """
 
         if isinstance(sg_response, dict) and sg_response.get("exception"):
-            raise Fault(sg_response.get("message",
-                "Unknown Error"))
+            
+            # handle the following cases explicitly
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Can't authenticate session token 'xxxxxxxx'"}
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Session token 'xxxxxxxx has expired'"}
+            
+            msg = sg_response.get("message")
+            
+            if msg and ("Can't authenticate session token" in msg or "has expired" in msg):
+                raise SessionAuthenticationFault(msg)
```

Comment:
Well, the ERR_AUTH will be passed back for all authentication issues, where as I was trying to get a special exception raised when a session has expired. Hence the string matching. But maybe it's enough to determine that it's an authentication issue? (e.g. bad username/password/permissions related/sessions etc). As basing it on the error code is safer and nicer, I'll make those changes first and then we can see if we need further refinement later down the line....